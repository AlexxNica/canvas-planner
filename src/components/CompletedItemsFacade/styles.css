.root {
  font-family: var(--fontFamily);
  color: var(--color);
  box-sizing: border-box;
  padding: var(--padding);
  border-bottom: var(--borderWidth) solid var(--borderColor);
}

.contentPrimary {
  margin-bottom: var(--bottomMarginPhoneUp);
  box-sizing: border-box;
  min-width: 1px;
}

.contentSecondary {
  box-sizing: border-box;
  min-width: 1px;
}

.button {
  user-select: none;
  touch-action: manipulation;
  box-sizing: border-box;
  font-family: var(--fontFamily);
  display: inline-flex;
  align-items: center;
  background: transparent;
  border: none;
  padding: var(--buttonPadding);
  cursor: pointer;
  outline: none;
  font-size: var(--buttonFontSize);
  color: var(--buttonColor);

  &:focus,
  &:hover {
    text-decoration: var(--buttonTextDecorationHover);
  }
}

.buttonCheckbox {
  box-sizing: border-box;
  width: 1.375rem;
  margin-right: var(--checkboxRightMarginPhoneUp);
  line-height: 1;
}

[data-media-medium] {
  &.root {
    padding: var(--paddingMedium);
    display: flex;
    align-items: center;
  }

  .contentPrimary {
    margin-bottom: 0;
    flex: 1;
  }

  .contentSecondary {
    margin-left: 1rem;
  }
}

[data-media-x-large] {
  &.root {
    padding: var(--paddingLarge);
  }

  .buttonCheckbox {
    margin-right: var(--checkboxRightMarginWideUp);
  }
}
