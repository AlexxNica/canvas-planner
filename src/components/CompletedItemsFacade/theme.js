/* Global variables (colors, typography, spacing, etc.) are defined in lib/themes */

export default function generator ({ borders, colors, media, spacing, typography }) {
  return {
    fontFamily: typography.fontFamily,
    color: colors.licorice,
    paddingPhoneUp: `${spacing.small} 0`,
    paddingTabletUp: `${spacing.small}`,
    paddingWideUp: `${spacing.small} ${spacing.medium}`,
    borderWidth: borders.widthSmall,
    borderColor: colors.tiara,
    bottomMarginPhoneUp: spacing.xSmall,
    buttonPadding: `${spacing.xSmall} 0`,
    buttonFontSize: typography.fontSizeMedium,
    buttonTextDecorationHover: 'underline',
    buttonColor: colors.brand,
    checkboxRightMarginPhoneUp: spacing.small,
    checkboxRightMarginWideUp: spacing.medium,

    ...media
  };
}
