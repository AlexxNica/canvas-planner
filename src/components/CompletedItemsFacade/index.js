import React, { Component } from 'react';
import themeable from 'instructure-ui/lib/themeable';
import CheckboxFacade from 'instructure-ui/lib/components/Checkbox/CheckboxFacade';
import { func, number } from 'prop-types';

import styles from './styles.css';
import theme from './theme.js';

import formatMessage from '../../format-message';

class CompletedItemsFacade extends Component {

  static propTypes = {
    onClick: func.isRequired,
    itemCount: number.isRequired
  }

  render () {
    return (
      <li className={styles.item}>
        <div className={styles.contentPrimary}>
          <button
            type="button"
            className={styles.button}
            onClick={this.props.onClick}
          >
            <span className={styles.buttonCheckbox} aria-hidden="true">
              <CheckboxFacade checked={true}>{''}</CheckboxFacade>
            </span>
            <span>
              {
                formatMessage(`{
                  count, plural,
                  one {Show # completed item}
                  other {Show # completed items}
                }`, { count: this.props.itemCount })
              }
            </span>
          </button>
        </div>
        <div className={styles.contentSecondary}>
          Pills
        </div>
      </li>
    );
  }
}

export default themeable(theme, styles)(CompletedItemsFacade);
