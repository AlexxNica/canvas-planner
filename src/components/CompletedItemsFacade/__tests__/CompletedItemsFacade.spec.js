import React from 'react';
import { shallow } from 'enzyme';
import CompletedItemsFacade from '../index';

it('renders as an li with a CheckboxFacade and a string of text indicating count', () => {
  const wrapper = shallow(
    <CompletedItemsFacade
      onClick={() => {}}
      itemCount={3}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('calls the onClick prop when clicked', () => {
  const fakeOnClick = jest.fn();
  const wrapper = shallow(
    <CompletedItemsFacade
      onClick={fakeOnClick}
      itemCount={0}
    />
  );

  wrapper.find('button').simulate('click');
  expect(fakeOnClick).toHaveBeenCalled();
});
