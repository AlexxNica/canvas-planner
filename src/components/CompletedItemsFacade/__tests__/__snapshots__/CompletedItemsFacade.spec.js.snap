// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as an li with a CheckboxFacade and a string of text indicating count 1`] = `
<li>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        <CheckboxFacade
          checked={true}
          focused={false}
          hovered={false}
          size="medium"
        />
      </span>
      <span>
        Show 3 completed items
      </span>
    </button>
  </div>
  <div>
    Pills
  </div>
</li>
`;
